src/databricks/labs/ucx/hive_metastore/locations.py

```diff
@@ -154,17 +154,60 @@ def __init__(
         schema: str,
         tables_crawler: TablesCrawler,
         mounts_crawler: 'MountsCrawler',
+        enable_hms_federation: bool = False,
     ):
         super().__init__(sql_backend, "hive_metastore", schema, "external_locations", ExternalLocation)
         self._ws = ws
         self._tables_crawler = tables_crawler
         self._mounts_crawler = mounts_crawler
+        self._enable_hms_federation = enable_hms_federation
 
     @cached_property
     def _mounts_snapshot(self) -> list['Mount']:
         """Returns all mounts, sorted by longest prefixes first."""
         return sorted(self._mounts_crawler.snapshot(), key=lambda _: (len(_.name), _.name), reverse=True)
 
+    @staticmethod
+    def clean_location(location: str) -> str:
+        # remove the s3a scheme and replace it with s3 as these can be considered the same and will be treated as such
+        # Having s3a and s3 as separate locations will cause issues when trying to find overlapping locations
+        return re.sub(r"^s3a:/", r"s3:/", location).rstrip("/")
+
+    def external_locations_with_root(self) -> list[ExternalLocation]:
+        # Returns a list of external locations with the DBFS root location appended to the list
+        # Used for HMS Federation use cases
+        external_locations = list(self.snapshot())
+        if not self._enable_hms_federation:
```

Comment:
I would remove it here as it is already covered by the `_get_dbfs_root`

Reply:
OK